notebooks/MAST/HSC/HSC_TAP/HSC_TAP.ipynb

```diff
@@ -0,0 +1,472 @@
+    "    - [Use Case: Plotting a light curve for the most variable object in a field](#Use-Case:-Plotting-a-light-curve-for-the-most-variable-object-in-a-field)\n",
+    "    - [Use Case: Create a color magnitude diagram for the Small Magellanic Cloud](#Use-Case:-Create-a-color-magnitude-diagram-for-the-Small-Magellanic-Cloud) <br><br>\n",
+    "5. [Additional Resources](#Additional-Resources)\n",
+    "6. [About This Notebook](#About-this-Notebook)\n",
+    "***"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## HSC TAP Service Introduction\n",
+    "<div style=\"text-align: right\">[(top)](#title_ID)</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Table Access Protocol (TAP) services allow more direct and flexible access to astronomical data than the simpler types of IVOA standard data services. Queries are built with the SQL-like Astronomical Data Query Language (ADQL), and can include geographic / spatial queries as well as filtering on other characteristics of the data. This also allows the user fine-grained control over the returned columns, unlike the fixed set of coumns retunred from cone, image, and spectral services.\n",
+    "\n",
+    "For this example, we'll be using the astroquery TAP/TAP+ client, which was developed by the ESAC Space Data Centre for working with the GAIA catalog, but is interoperable with other valid TAP services, including those at MAST. As an astroquery project, TAP+ documentation is available at ReadTheDocs: http://astroquery.readthedocs.io/en/latest/utils/tap.html\n",
+    "\n",
+    "We'll be using TAP+ to call the most recent version (3) of the Hubble Source Catalog TAP service at MAST. The schema is described within the service, and we'll show how to inspect it. The schema is also the same as the one available via the CasJobs interface, with an additional view added for the most common positional queries. CasJobs has its own copy of the schema documentation, which can be accessed through its own site: http://mastweb.stsci.edu/hcasjobs/\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "***\n",
+    "## Imports\n",
+    "<div style=\"text-align: right\">[(top)](#title_ID)</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Use the astroquery TapPlus library as our client to the data service.\n",
+    "from astroquery.utils.tap.core import TapPlus\n",
+    "\n",
+    "## For handling ordinary astropy Tables in responses\n",
+    "from astropy.table import Table, vstack\n",
+    "\n",
+    "## For reading FITS files once we locate them via TAP\n",
+    "import astropy.io.fits as apfits\n",
+    "\n",
+    "# For displaying and manipulating some types of results\n",
+    "%matplotlib inline\n",
+    "import requests, io, astropy, numpy, pylab, time\n",
```

Comment:
A few things here:

* PEP8 style says imports should generally be on one line - e.g.:
```
import io
import requests
import numpy
```
* Because it's so commonly-used, the recommended convention is ``import numpy as np`` and then do ``np.<whatever>`` when you use numpy
* relatedly, the `pylab` interface is no longer recommended.  Instead you should do `from matplotlib import pyplot as plt` and then `plt.<whatever>
* `io` isn't used anywhere in the notebook, so shouldn't import it

Reply:
Changed to match PEP8 style, also numpy as np and change to pyplot. removed io.